spring-kafka/src/main/java/org/springframework/kafka/listener/ListenerUtils.java

```diff
@@ -185,11 +187,22 @@ public static void unrecoverableBackOff(BackOff backOff, ThreadLocal<BackOffExec
 	 * @since 2.7
 	 */
 	public static void stoppableSleep(MessageListenerContainer container, long interval) throws InterruptedException {
+		stoppableSleep(container::isRunning, interval);
+	}
+
+	/**
+	 * Sleep for the desired timeout, as long as shouldSleepCondition supplies true.
+	 * @param shouldSleepCondition to.
+	 * @param interval the timeout.
+	 * @throws InterruptedException if the thread is interrupted.
+	 * @since 3.0.9
```

Comment:
@antonio-tomac Are you using 3.0.x? If so, we won't need to back port this feature. If you are still using 2.9.x, we should make the since 2.9.10.

Reply:
using 3.0.7 but plan to upgrade to 3.0.9 ASAP

Reply:
If you want backport, I can put since 2.9.10, but I don't need backport myself

Reply:
No; if you don't need it; I'd rather not back port.